nussl/audio_signal.py

```diff
@@ -108,16 +109,34 @@ def __str__(self):
     ##################################################
     #                   Plotting
     ##################################################
-
-    # def plot_time_domain(self):
-    #     """
-    #     Not implemented yet -- will raise and exception
-    #     Returns:
     #
-    #     """
-    #     signal = self.audio_data
-    #     for channel in num_channels
 
+    def plot_time_domain(self, x_label='time', save=False, name=None, output_path=None):
+        """
+        Parameters:
+            x_label (str): Label the x axis with time or samples
+            save (bool): Save a png of the plot to the current folder
+            name (str): The name of the audio signal. Applies to title and filename
+            output_path (str): The output path of where the plot is saved if save is True
+
+        """
+        signal = self.audio_data
+        for i in range(self.num_channels):
+            plt.subplot(self.num_channels+1, 1, i+1)
+            if x_label == 'time':
+                x_axis = np.linspace(0, len(signal[i])/self.sample_rate, len(signal[i]))
```

Comment:
There is already a property that does exactly what this line does. Look at `self.time_vector`.